### PR TITLE
Added a complete solution for volumetric clouds for HDRP including a cloud map generation tool.

### PR DESCRIPTION
**Pull request by anisunity**
This PR adds a solution for volumetric clouds in HDRP. The tool can be found under the "Volumetric Clouds" volume component.

image

Note that most of the prameters can be found when expanding into the advanced mode.

image

The cloud map generation tool should be used for the generation

image

The feature is different from the cloud layers as the clouds are in the scene and can be navigated throught.

image

image

The feature has been pre-tested by QA and artists and will still be.

Automated tests still need to be added.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Lighting/VolumetricClouds/VolumetricCloudsWindow.cs
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/VolumetricClouds/VolumetricCloudsWindow.cs:180]: needs to be removed

**Comment by andrew-raphael-lukasik** [comment]
Dear Santa,

One thing I would love this year is:
URP Clouds.

Xoxo

**Comment by SeanPuller** [comment]
Issues as discussed 
1. Default preset texture not working for "Simple" after switching scenes
2. Reprojection not working at arbitrary resolutions

**Comment by anisunity** [comment]
@SeanPuller both reported issues have been fixed

**Comment by anisunity** [comment]
Added the support of a shadow 
image

**Review by remi-chapelain — COMMENTED** [review]
Not adding review yet but [linking Test document here](https://confluence.unity3d.com/display/HDRP/Volumetrics+clouds) and the summary for awareness : 

Testing still in progress but,

PR is getting there in terms of visual quality and bug count. 

Performance wise, the PR is completely usable as is and doesn't cost much, there's a higher cost at higher resolution but nothing unexpected. 

Remaining issues are (slightly ordered) : 
- It's still broken with most platforms (Metal, Vulkan, PS4, Xbox)
- Support for planar/reflection probe needs to be added
- Support for Ray Traced reflections needs to be added
- HDRP Asset and HDRP Default frame settings needs to be added
- Quality levels presets needs to be added to make the use easier for the user
- UI / Tooltip needs a quick feedback/review from testing team (artists + qa)
- Documentation needs to be added
- Volume parameters needs to be able to be update at runtime

**Comment by kroatoa** [comment]
What is the reason to wait with the automated tests? How is adding these tracked?

**Comment by kroatoa** [comment]
> 
> 
> What is the reason to wait with the automated tests? How is adding these tracked?

I got confirmation over slack that the tests are being added as part of this PR.

**Comment by RemyUnity** [comment]
I think the _(Preview)_ label in not necessary and should be removed.

**Review by kroatoa — APPROVED** [review]

**Review by remi-chapelain — APPROVED** [review]
After discussing with @JulienIgnace-Unity, @anisunity and @RemyUnity, we agreed that the best course of action is approving as is and try to fix the missing platforms before these changes lands in Beta. 
This will avoid spending too much time maintaining the PR compatible with latest changes and with SRP2Core, we'll be able to get some user feedback from alpha versions. 

Here's the [final test document](https://confluence.unity3d.com/display/HDRP/Volumetrics+clouds).
 
Here's what needs fixing before beta : 
- Metal Support
- XboxOne Support

And here's what can be added/fixed in the near future 
- Multi-camera shadows
- Slight jump on clouds when entering a volume with volumetric fog
- Support for Planar / Probe / Ray Tracing
- Some Parameters like (Wind orientation, clouds altitude.. ) still aren't suited for blending properly
- Quality settings for primary steps and light steps

**Review by JulienIgnace-Unity — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs.hlsl
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs.hlsl:43]: something is wrong here in code generation

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs:97]: questoin:  does it need to be R16G16B16A16_SFloat or can it be 111110f?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs:255]: note follow the conventoin for constant => k_

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs:595]: all the lighting for cloud, isn't this suppose to reuse the current color buffer format?
GraphicsFormat.R16G16B16A16_SFloat
Guess we could follow it, or maybe give the option to use 111110f in the settings? it is a lot of fat texture.

File: com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs.hlsl
Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.cs.hlsl:26]: Bad merge here

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs
Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs:97]: No can't, alpha channel is used

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs
Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs]: ok

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs
Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs:595]: Not it cannot, the alpha channel is used

**Pull request merged**
